src/NuGet.Core/NuGet.Frameworks/NuGetFramework.cs

```diff
@@ -57,8 +57,34 @@ public NuGetFramework(string frameworkIdentifier, Version frameworkVersion, stri
 
             _frameworkIdentifier = frameworkIdentifier;
             _frameworkVersion = NormalizeVersion(frameworkVersion);
-            _frameworkProfile = frameworkProfile ?? string.Empty;
+
+            IsNet5Era = (_frameworkVersion.Major >= Version5 && StringComparer.OrdinalIgnoreCase.Equals(FrameworkConstants.FrameworkIdentifiers.NetCoreApp, _frameworkIdentifier));
+
+            _frameworkProfile = profile ?? string.Empty;
+            Platform = string.Empty;
+            PlatformVersion = FrameworkConstants.EmptyVersion;
+        }
+
+        public NuGetFramework(string frameworkIdentifier, Version frameworkVersion, string platform, Version platformVersion)
```

Comment:
The logic for this constructor depends on the IsNet5Era. I think that method comments would be useful for the consumers.